torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -2485,6 +2488,9 @@ class IrParser {
                    static_cast<c10::TypePtr>(NoneType::get()))) {
       // TODO: should we consider adding support for NoneType;
       return true;
+    } else if (val->type()->isSubtypeOf(
+                   static_cast<c10::TypePtr>(StringType::get()))) {
+      return true;
```

Comment:
Can you put a note here that `string` scalars are only used in parsing rule, hence it's ok to not register it with codegen IR.